docs/install/osx_setup.md

```diff
@@ -97,17 +101,23 @@ Install the dependencies, required for MXNet, with the following commands:
 ### Build MXNet Shared Library
 After you have installed the dependencies, pull the MXNet source code from Git and build MXNet to produce an MXNet library called ```libmxnet.so```. You can clone the repository as described in the following code block, or you may try the [download links](download.md) for your desired MXNet version.
 
-The file called ```osx.mk``` has the configuration required for building MXNet on OS X. First copy ```make/osx.mk``` into ```config.mk```, which is used by the ```make``` command:
-
 ```bash
-    git clone --recursive https://github.com/apache/incubator-mxnet ~/mxnet
-    cd ~/mxnet
-    cp make/osx.mk ./config.mk
-    echo "USE_BLAS = openblas" >> ./config.mk
-    echo "ADD_CFLAGS += -I/usr/local/opt/openblas/include" >> ./config.mk
-    echo "ADD_LDFLAGS += -L/usr/local/opt/openblas/lib" >> ./config.mk
-    echo "ADD_LDFLAGS += -L/usr/local/lib/graphviz/" >> ./config.mk
-    make -j$(sysctl -n hw.ncpu)
+git clone --recursive https://github.com/apache/incubator-mxnet ~/mxnet
+cd ~/mxnet && pushd .
+mkdir -p build && cd build
+cmake \
+    -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
+    -DCMAKE_C_COMPILER_LAUNCHER=ccache \
+    -DUSE_MKL_IF_AVAILABLE=OFF \
+    -DUSE_MKLDNN=ON \
```

Comment:
I feel all these flags are unnecessary. We already have a CMakeList.txt and isn't that the place for user to switch on/off these flags?

Reply:
Usually users shouldn't be touching cmakelists.txt

Reply:
@apeforest that's the way CMake works. I'm not sure what exactly are you suggesting. Feedback should be concrete, timely and actionable to be useful.